File: GenerateAllSubarrays.c
#include <stdio.h>
int generateAllSubarrays(int arr[], int n) {
    // Duyệt qua tất cả các chỉ số bắt đầu
    for (int i = 0; i < n; i++) {
        // Duyệt qua tất cả các chỉ số kết thúc cho subarray từ i đến j
        for (int j = i; j < n; j++) {
            // In ra subarray từ i đến j
            for (int k = i; k <= j; k++) {
                printf("%d ", arr[k]);
            }
            printf("\n"); // In dòng mới sau mỗi subarray
        }
    }
    return 0;
}
int main() {
    int n;
    printf("Nhập số lượng phần tử của mảng: ");
    scanf("%d", &n);
    int arr[n];
    // Yêu cầu nhập các phần tử của mảng
    printf("Nhập các phần tử của mảng: ");
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }
    printf("Các subarray của mảng là:\n");
    generateAllSubarrays(arr, n);
    return 0;
}

File: GenerateAllSubarrays.py
def printSubArrays(arr):
    n = len(arr)
    
    # Duyệt qua tất cả các chỉ số bắt đầu của subarray
    for i in range(n):
        # Duyệt qua tất cả các chỉ số kết thúc của subarray
        for j in range(i, n):
            # In ra subarray từ i đến j
            print(arr[i:j+1])

# Yêu cầu người dùng nhập mảng
arr_input = input("Nhập các phần tử của mảng cách nhau bởi khoảng trắng: ")

# Chuyển chuỗi nhập vào thành danh sách các số nguyên
arr = list(map(int, arr_input.split()))

# Gọi hàm để in ra tất cả các subarray
print("Các subarray của mảng là:")
printSubArrays(arr)
